src/test/java/org/opensearch/flowframework/workflow/CreateIndexStepTests.java

```diff
@@ -10,54 +10,88 @@
 import org.opensearch.core.action.ActionListener;
+import org.opensearch.flowframework.indices.FlowFrameworkIndex;
 import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.threadpool.ThreadPool;
 
-import java.io.IOException;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
 
-import static org.mockito.Mockito.any;
+import static org.opensearch.flowframework.common.CommonValue.GLOBAL_CONTEXT_INDEX;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 public class CreateIndexStepTests extends OpenSearchTestCase {
 
-    private WorkflowData inputData = WorkflowData.EMPTY;
+    private static final String META = "_meta";
+    private static final String SCHEMA_VERSION_FIELD = "schemaVersion";
 
+    private WorkflowData inputData = WorkflowData.EMPTY;
     private Client client;
-
     private AdminClient adminClient;
-
+    private CreateIndexStep createIndexStep;
+    private ThreadContext threadContext;
+    private Metadata metadata;
+    private Map<String, AtomicBoolean> indexMappingUpdated = new HashMap<>();
+
+    @Mock
+    private ClusterService clusterService;
+    @Mock
     private IndicesAdminClient indicesAdminClient;
+    @Mock
+    private ThreadPool threadPool;
+    @Mock
+    IndexMetadata indexMetadata;
 
     @Override
     public void setUp() throws Exception {
         super.setUp();
-
+        MockitoAnnotations.openMocks(this);
         inputData = new WorkflowData(Map.ofEntries(Map.entry("index-name", "demo"), Map.entry("type", "knn")));
+        clusterService = mock(ClusterService.class);
         client = mock(Client.class);
         adminClient = mock(AdminClient.class);
-        indicesAdminClient = mock(IndicesAdminClient.class);
+        metadata = mock(Metadata.class);
+        Settings settings = Settings.builder().build();
+        threadContext = new ThreadContext(settings);
 
-        when(adminClient.indices()).thenReturn(indicesAdminClient);
+        when(client.threadPool()).thenReturn(threadPool);
+        when(threadPool.getThreadContext()).thenReturn(threadContext);
         when(client.admin()).thenReturn(adminClient);
+        when(adminClient.indices()).thenReturn(indicesAdminClient);
+        when(clusterService.state()).thenReturn(ClusterState.builder(new ClusterName("test cluster")).build());
+        when(metadata.indices()).thenReturn(Map.of(GLOBAL_CONTEXT_INDEX, indexMetadata));
 
+        createIndexStep = new CreateIndexStep(clusterService, client);
```

Comment:
Can we mock CreateIndexStep completely?

Reply:
Not sure what you mean exactly.. If we're write unit tests for class `CreateIndexStep`, shouldn't we instantiate the actual class, mock its dependencies, and test its behavior?